openshift_cli_installer/libs/clusters/ipi_cluster.py

```diff
@@ -105,23 +113,37 @@ def _create_install_config_file(self):
         if fips:
             terraform_parameters["fips"] = fips
 
-        cluster_install_config = get_install_config_j2_template(jinja_dict=terraform_parameters)
+        cluster_install_config = get_install_config_j2_template(jinja_dict=terraform_parameters, platform=platform)
 
         with open(os.path.join(self.cluster_info["cluster-dir"], "install-config.yaml"), "w") as fd:
             fd.write(yaml.dump(cluster_install_config))
 
     def _set_install_version_url(self):
         cluster_version = self.cluster["version"]
-        version_url = [url for url, versions in self.aws_base_available_versions.items() if cluster_version in versions]
+        version_url = [url for url, versions in self.ipi_base_available_versions.items() if cluster_version in versions]
         if version_url:
             self.cluster_info["version-url"] = f"{version_url[0]}:{self.cluster_info['version']}"
         else:
             self.logger.error(
                 f"{self.log_prefix}: Cluster version url not found for"
-                f" {cluster_version} in {self.aws_base_available_versions.keys()}",
+                f" {cluster_version} in {self.ipi_base_available_versions.keys()}",
             )
             raise click.Abort()
 
+    def get_gcp_project_id(self):
+        with open(self.gcp_service_account_file, "r") as fd:
+            gcp_sa_file_content = json.load(fd)
+
+        return gcp_sa_file_content["project_id"]
+
+    def save_gcp_service_account_file(self):
+        gcp_file_dir = os.path.join(os.getenv("HOME"), ".gcp")
+        Path(gcp_file_dir).mkdir(parents=True, exist_ok=True)
+
+        gcp_file_path = os.path.join(gcp_file_dir, "osServiceAccount.json")
+        self.logger.info(f"Saving GCP ServiceAccount file to {gcp_file_path}")
+        shutil.copy(self.gcp_service_account_file, gcp_file_path)
```

Comment:
you need to check if a user has a file in this dir - if a file exists you need to save and restore it after installer is done

Reply:
@rnetser updated the code, please review